src/coniql/app.py

```diff
@@ -1,49 +1,75 @@
-import traceback
 from argparse import ArgumentParser
 from pathlib import Path
-from typing import Any, Dict
+from typing import List, Optional
 
 import aiohttp_cors
+import strawberry
 from aiohttp import web
-from tartiflette import Engine, TartifletteError
-from tartiflette_aiohttp import register_graphql_handlers
+from graphql import GraphQLError
+from strawberry.aiohttp.views import GraphQLView
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL
+from strawberry.types import ExecutionContext
 
-from coniql.caplugin import CAPlugin
-from coniql.plugin import PluginStore
-from coniql.pvaplugin import PVAPlugin
-from coniql.simplugin import SimPlugin
+import coniql.strawberry_schema as schema
 
 from . import __version__
 
 
-async def error_coercer(exception: Exception, error: Dict[str, Any]) -> Dict[str, Any]:
-    if isinstance(exception, TartifletteError):
-        e = exception.original_error
+class ConiqlSchema(strawberry.Schema):
+    # Override to remove stack trace printer with every log
+    def process_errors(
+        self,
+        errors: List[GraphQLError],
+        execution_context: Optional[ExecutionContext] = None,
+    ) -> None:
+        for error in errors:
+            print(error.message)
```

Comment:
I tested out the mechanism described in the link and unfortunately it doesn't do what we want it to do. It allows you to intercept the graphql result and any errors before you send it over the websocket, which means that you can manipulate the error sent to the client. However it doesn't have any affect on what is logged as part of the Coniql application, i.e. the stack trace always gets printed in the Coniql terminal.

The implementation I currently have is the only way I can find to give us the option to manipulate what is actually logged and hide the stack trace in the deployed system.

Reply:
Fair enough. I'd like to see a note that this is undocumented, and therefore may break with Strawberry updates without notice. Just so the next developer has some idea why it may suddenly be broken.

Reply:
Added a comment about this in commit: https://github.com/dls-controls/coniql/pull/44/commits/21986ea61570d37ee21a8529aa645f70623506c9